Helper/Quote.php

```diff
@@ -130,4 +139,60 @@ public function getIsQuoteMultiShippingWithMerchantReference(string $merchantRef
 
         return $quote->getIsMultiShipping();
     }
+
+    /**
+     * Get inactive quote for user. This function is very similar to GetCartForUser::execute.
+     *
+     * @param string $cartHash
+     * @param int|null $customerId
+     * @param int $storeId
+     * @return QuoteModel
+     * @throws NoSuchEntityException
+     */
+    public function getInactiveQuoteForUser(string $cartHash, ?int $customerId, int $storeId): QuoteModel
+    {
+        try {
+            $cartId = $this->maskedQuoteIdToQuoteId->execute($cartHash);
+        } catch (NoSuchEntityException $exception) {
+            throw new NoSuchEntityException(
+                __('Could not find a cart with ID "%masked_cart_id"', ['masked_cart_id' => $cartHash])
+            );
+        }
+
+        try {
+            /** @var QuoteModel $cart */
+            $cart = $this->cartRepository->get($cartId);
+        } catch (NoSuchEntityException $e) {
+            throw new NoSuchEntityException(
+                __('Could not find a cart with ID "%masked_cart_id"', ['masked_cart_id' => $cartHash])
+            );
+        }
+
+        if ($cart->getIsActive()) {
```

Comment:
Why do we throw an exception if a cart is active?

Reply:
Because it shouldn't be active since these calls will be made after the order has been completed

Reply:
ok got it